workspace-service/app/services/impl/docker_workspace_service.go

```diff
@@ -76,11 +79,25 @@ func (ws DockerWorkspaceService) checkAndCreateWorkspaceFromTemplate(workspaceId
 
 	ws.logger.Info("Creating workspace from template", zap.String("workspaceId", workspaceId), zap.String("backendTemplate", backendTemplate))
 
+	//copying backend template in root dir
 	err = utils.SudoRsyncFolders("/templates/"+backendTemplate+"/", "/workspaces/"+workspaceId)
 	if err != nil {
 		ws.logger.Error("Failed to rsync folders", zap.Error(err))
 		return err
 	}
+	//creating a frontend folder in the directory
+	frontendPath := "/workspaces/"+workspaceId+"/frontend"
```

Comment:
Check if it's optional and skip creation.

Reply:
made the frontendTemplate optional, only if it not nil then only the frontend template will be copied